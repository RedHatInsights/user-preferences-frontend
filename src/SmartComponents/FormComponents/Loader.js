import React from 'react';
import { Skeleton } from '@redhat-cloud-services/frontend-components/Skeleton';
import PropTypes from 'prop-types';

const Loader = ({
  name = '', // eslint-disable-line no-unused-vars
  size = 'md',
  FieldProvider,
  validate,
  FormSpyProvider,
  formOptions,
  ...rest
}) => (
  <div {...rest}>
    <Skeleton size={size}></Skeleton>
  </div>
);

Loader.propTypes = {
  FieldProvider: PropTypes.any,
  formOptions: PropTypes.any,
  FormSpyProvider: PropTypes.any,
  validate: PropTypes.any,
  name: PropTypes.string,
  size: PropTypes.string,
};

export default Loader;
