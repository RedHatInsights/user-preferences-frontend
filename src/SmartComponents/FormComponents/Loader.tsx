import React from 'react';
import { Skeleton } from '@redhat-cloud-services/frontend-components/Skeleton';

export type LoaderProps = {
  FieldProvider: any,
  formOptions: any,
  FormSpyProvider: any,
  validate: any,
  name: string,
  size: any,
}

const Loader = ({
  name = '',
  size = 'md',
  FieldProvider,
  validate,
  FormSpyProvider,
  formOptions,
  ...rest
}: LoaderProps) => {
  return (<div {...rest}>
    <Skeleton size={size}></Skeleton>
  </div>)
}

export default Loader;
