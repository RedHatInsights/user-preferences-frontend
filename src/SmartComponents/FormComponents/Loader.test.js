import { render } from '@testing-library/react';
import React from 'react';

import Loader from './Loader';

describe('Loader component tests', () => {
  it('should render correctly', () => {
    const { container } = render(<Loader size="sm"></Loader>);
    expect(container).toMatchSnapshot();
  });
});
