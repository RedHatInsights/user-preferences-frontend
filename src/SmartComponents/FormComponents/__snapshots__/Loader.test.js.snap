// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component tests should render correctly 1`] = `
<div>
  <div>
    <div
      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
    >
      <span
        class="pf-u-screen-reader"
      />
    </div>
  </div>
</div>
`;
